### PR TITLE
scopegen: add option to use output version 2

### PR DESCRIPTION
**Pull request by favadi**
output version 2 has following changes:

* all custom types ServiceScope/AuthScope removed, as they are used in no where. Consumes are converting them to map[string]string everywhere
* keeping scope names in generated codes

### CODE REVIEW

File: scopegen/generate/json.go [hunk 1]
```diff
@@ -2,18 +2,54 @@ package generate
...
 	}
 	return json.NewEncoder(w).Encode(result)
 }
+
+type model struct {
+	Scopes     map[string]string   `json:"scopes"`
+	AuthScopes map[string][]string `json:"auth_scopes"`
+}
+
+// jsonV2 generates mapping between service full method names and its OAuth2 scope in JSON format.
```
Comment (rnd) [inline scopegen/generate/json.go]: nit: maybe explain what the difference between jsonV1 and jsonV2 format on the doc here.
summary of what you have on commit message i think it's ok

File: scopegen/scopegen.go [hunk 2]
```diff
@@ -51,6 +53,11 @@ func (s *scopePlugin) Generate(req *plugin_go.CodeGeneratorRequest) (*plugin_go.
 				}
 			case "out":
 				baseDir = v
+			case "output_version":
+				var err error
+				if outputVersion, err = strconv.Atoi(v); err != nil {
+					return nil, fmt.Errorf("unknown output version: err=%s", err.Error())
```
Comment (mgfcmartinez) [inline scopegen/scopegen.go]: nit: not `:%w`?

File: testdata/scripts/scopegen_no_swagger_v2.txt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+// Message comment
+type Message struct {
+	Name string `pb:"1" json:"name"`
+}
+
+type Service interface {
+	GetMessage(Message) Message
+}
+-- all.scopes.json.golden --
```
Comment (mgfcmartinez) [inline testdata/scripts/scopegen_no_swagger_v2.txt]: out of scope for the PR but why do we generate the scope files when the definitions don't contain scopes?

**Comment by mgfcmartinez** [comment]
ah, also `Consumers` in the commit message

File: scopegen/scopegen.go (hunk 2 above)
Comment (favadi) [reply scopegen/scopegen.go]: `%w` is too new, we are still support go 1.11.

File: testdata/scripts/scopegen_no_swagger_v2.txt (hunk 3 above)
Comment (favadi) [reply testdata/scripts/scopegen_no_swagger_v2.txt]: It is clearer/more convenience to output an empty scopegen file than omit it. No file generated could mean many things, an empty file clearly told user that the file has successfully generated.

File: scopegen/generate/json.go (hunk 1 above)
Comment (favadi) [reply scopegen/generate/json.go]: I added a new section in README.md.

**Pull request merged**